packages/next/src/build/index.ts

```diff
@@ -2106,7 +2106,8 @@ export default async function build(
               denormalizedAppPages,
               outputFileTracingRoot,
               requiredServerFiles.config,
-              middlewareManifest
+              middlewareManifest,
+              Boolean(config.experimental.instrumentationHook)
```

Comment:
When I was looking at this yesterday I was thinking you want to pass hasInstrumentationHook (i.e. whether the file exists), not the config value here. Otherwise it will probably crash if the flag is enabled but no instrumentation set up?

Reply:
That's what I get for not writing proper tests.
Thanks for finding this.

Reply:
This should be fixed now.